rfcs/convergence/contract-for-trigger-pattern.md

```diff
@@ -0,0 +1,134 @@
+# RFC: Creating a structured contract for Trigger - like patterns
+
+@ling1726
+
+## Problem statement
+
+```tsx
+<Tooltip>
+  <Button />
+</Tooltip>
+
+<Menu>
+  <MenuTrigger>
+    <Button />
+  </MenuTrigger>
+</Menu>
+
+<Popover>
+  <PopoverTrigger>
+    <Button />
+  </PopoverTrigger>
+</Popover>
+```
+
+The above examples show the adoption of a `Trigger` pattern that was inspired from [react-spectrum](https://react-spectrum.adobe.com/react-spectrum/MenuTrigger.html), where the library component will clone the child and add appropriate event handlers/props to handle complex interactions to simplify the lives of our users. These can be:
+
+- `aria-*` and `id` attributes to ensure and a11y friendly experience
+- `onClick` `onMouseEnter` `onFocus` event handlers
+
+In the case of `Menu` and `Popover`. There is not really an easy way to know for consumers what exactly is being passed down to children components if they want to create their own components. `Tooltip` has been a great pioneer in this direction by introducing the possibility of using a render prop:
+
+```tsx
+<Tooltip content="The trigger button was rendered by a render function" triggerAriaAttribute="describedby">
+  {triggerProps => (
+    <div>
+      <button {...triggerProps}>Custom trigger</button>
+    </div>
+  )}
+</Tooltip>
+```
+
+Render props are not the most ergonomic to use and make optimizing the child with `PureComponent` or `memo` harder. since the child which is a function in this case will always be shallow compared. and `PureComponents` inside the render prop will be recreated.
```

Comment:
Can you please add a CodeSandbox with a demo? I didn't get the problem there: components will be not be recreated, check a CodeSandbox: https://codesandbox.io/s/vigilant-heisenberg-8i1vi?file=/src/App.js